File: js/src/utils/getReverseKeyFromDomainKey.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getReverseKeyFromDomainKey = void 0;
const constants_1 = require("../constants");
const getHashedNameSync_1 = require("./getHashedNameSync");
const getNameAccountKeySync_1 = require("./getNameAccountKeySync");
/**
 * This function can be used to get the reverse key from a domain key
 * @param domainKey The domain key to compute the reverse for
 * @param parent The parent public key
 * @returns The public key of the reverse account
 */
const getReverseKeyFromDomainKey = (domainKey, parent) => {
    const hashedReverseLookup = (0, getHashedNameSync_1.getHashedNameSync)(domainKey.toBase58());
    const reverseLookupAccount = (0, getNameAccountKeySync_1.getNameAccountKeySync)(hashedReverseLookup, constants_1.REVERSE_LOOKUP_CLASS, parent);
    return reverseLookupAccount;
};
exports.getReverseKeyFromDomainKey = getReverseKeyFromDomainKey;

File: python/src/utils/get_reverse_key_from_domain_key.py
from typing import Optional

from solders.pubkey import Pubkey

from constants import REVERSE_LOOKUP_CLASS
from utils.get_hashed_name import get_hashed_name
from utils.get_name_account_key import get_name_account_key


def get_reverse_key_from_domain_key(
    domain_key: Pubkey, parent: Optional[Pubkey] = None
) -> Pubkey:
    """
    Get the reverse lookup key from the domain key.
    :param domain_key:
    :param parent:
    :return:
    """
    hashed_reverse_lookup = get_hashed_name(str(domain_key))
    return get_name_account_key(  # type: ignore[no-any-return]
        hashed_reverse_lookup, name_class=REVERSE_LOOKUP_CLASS, name_parent=parent
    )
